platform/ui/src/components/downloadDialog/DownloadDialog.js

```diff
@@ -0,0 +1,279 @@
+import React, { useEffect, useState } from 'react';
+import Modal from 'react-bootstrap-modal';
+
+import './DownloadDialog.styl';
+import { TextInput, Select } from '@ohif/ui';
+import { withTranslation } from '../../utils/LanguageProvider';
+
+const FILE_TYPE_OPTIONS = [
+  {
+    key: 'jpg',
+    value: 'jpg'
+  },
+  {
+    key: 'png',
+    value: 'png'
+  }
+];
+
+const DownloadDialog = ({
+  t,
+  isOpen,
+  activeViewport,
+  onClose,
+  updateViewportPreview,
+  enableViewport,
+  disableViewport,
+  toggleAnnotations,
+  loadImage,
+  downloadBlob,
+  defaultSize,
+  minimumSize,
+  canvasClass
+}) => {
+  const [filename, setFilename] = useState('image');
+  const [fileType, setFileType] = useState('jpg');
+
+  const [height, setHeight] = useState(defaultSize);
+  const [width, setWidth] = useState(defaultSize);
+
+  const [showAnnotations, setShowAnnotations] = useState(true);
+
+  const [keepAspect, setKeepAspect] = useState(true);
+  const [lastImage, setLastImage] = useState();
+
+  const [viewportElement, setViewportElement] = useState();
+  const [viewportElementHeight, setViewportElementHeight] = useState(minimumSize);
+  const [viewportElementWidth, setViewportElementWidth] = useState(minimumSize);
+
+  const [downloadCanvas, setDownloadCanvas] = useState();
+  const [downloadCanvasHeight, setDownloadCanvasHeight] = useState(minimumSize);
+  const [downloadCanvasWidth, setDownloadCanvasWidth] = useState(minimumSize);
+
+  const [viewportPreview, setViewportPreview] = useState();
+  const [viewportPreviewSrc, setViewportPreviewSrc] = useState();
+  const [viewportPreviewHeight, setViewportPreviewHeight] = useState(minimumSize);
+  const [viewportPreviewWidth, setViewportPreviewWidth] = useState(minimumSize);
```

Comment:
Love hooks. My only comment is that any state that you often update together can be grouped as a single `useState` with an object + properties. I think it's _slightly_ more performant and less code.

Reply:
Done! thx.